### ISSUE TITLE
Wait for RnR to be Consistently Available

### ISSUE THREAD

**Issue by mfulgo** [IBM-Watson/kale#59]
When creating an Retrieve and Rank cluster, kale should wait for the cluster to be consistently available. Not doing so creates an edge-case scenario in which `kale assemble kale-test english` may fail when attempting to create the solr-configuration:

```
...
[Running command 'kale create retrieve_and_rank kale-test-rnr']
Creating retrieve_and_rank service 'kale-test-rnr' using the 'standard' plan.
Creating key for service 'kale-test-rnr'.

Service 'kale-test-rnr' has been created and selected for future actions.

[Running command 'kale create cluster kale-test-cluster --wait']
Creating cluster 'kale-test-cluster' in 'kale-test-rnr'.
Waiting for cluster to become ready.
..................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

Cluster 'kale-test-cluster' has been created and selected for future actions.

[Running command 'kale create solr-configuration english']
Creating configuration 'english' in 'kale-test-rnr/kale-test-cluster'.
A call to: https://gateway.watsonplatform.net/retrieve-and-rank/api/v1/solr_clusters/<ID removed>/config/english
returned an unexpected error status code 404
{"msg":"WRRCSS030: Solr cluster ID [<ID removed>] does not exist.","code":404}

[An error occurred, starting rollback]
...
```

**Comment by cjgala** [comment]
Kale already does this.  The flag `--wait` is used to tell Kale to wait for the cluster to be available before doing anything else (which is used in the output you showed).

Unfortunately it's a not a perfect method.  Even if RnR says that a cluster is available, there is a small time window where the cluster is not available on all instances of RnR for a given tile.  This is a known issue with RnR, and there really isn't anything we can do on the Kale side of things.  I just poked the Retrieve team, and they're going to take another look at the issue.

Closing the issue as there is nothing we can do in Kale for this.

**Comment by mfulgo** [comment]
Per our discussion, kale can: ping/query RnR repeatedly for a period of time after it first says it's available, with some upper limit.

For instance, once it thinks the cluster is ready, kale could ping the cluster over the course of another 15 seconds to ensure none of the responses come back as "not ready".  If any of the requests say it's not ready, restart the 15 second timer unless the elapsed time has exceeded a minute.

**Comment by cjgala** [comment]
Resolved by #60
